packages/core/src/modules/oob/OutOfBandModule.ts

```diff
@@ -218,36 +218,63 @@ export class OutOfBandModule {
+    })
+    await this.outOfBandService.save(outOfBandRecord)
+
+    if (autoAcceptMessage) {
+      return await this.acceptMessage(outOfBandRecord, { label, autoAcceptConnection, reuseConnection })
+    }
+
+    return { outOfBandRecord }
+  }
+
+  public async acceptMessage(
+    outOfBandRecord: OutOfBandRecord,
+    config: {
+      autoAcceptConnection?: boolean
+      reuseConnection?: boolean
+      label?: string
+      mediatorId?: string
+    }
+  ) {
+    const { outOfBandMessage } = outOfBandRecord
+    const { label, autoAcceptConnection, reuseConnection } = config
+    const { handshakeProtocols, services } = outOfBandMessage
+    const messages = outOfBandMessage.getRequests()
+
     const existingConnection = await this.findExistingConnection(services)
 
     if (handshakeProtocols) {
       this.logger.debug('Out of band message contains handshake protocols.')
       // Find first supported handshake protocol preserving the order of `handshake_protocols`
       // in out-of-band message.
-      const handshakeProtocol = this.getFirstSupportedProtocol(handshakeProtocols)
 
-      let connectionRecord: ConnectionRecord
-      if (existingConnection) {
-        this.logger.debug('Connection already exists.', { connectionId: existingConnection.id })
-        if (reuseConnection) {
-          this.logger.debug('Reuse is enabled. Reusing an existing connection.')
-          connectionRecord = existingConnection
-          if (!messages) {
-            this.logger.debug('Out of band message does not contain any request messages.')
-            await this.sendReuse(outOfBandMessage, connectionRecord)
-          }
-        } else {
-          this.logger.debug('Reuse is disabled.')
-          connectionRecord = await this.createConnection(outOfBandMessage, { handshakeProtocol, autoAcceptConnection })
+      let connectionRecord
+      if (existingConnection && reuseConnection) {
+        this.logger.debug(`Reuse is enabled. Reusing an existing connection with ID ${existingConnection.id}.`)
+        connectionRecord = existingConnection
+        if (!messages) {
+          this.logger.debug('Out of band message does not contain any request messages.')
+          await this.sendReuse(outOfBandMessage, connectionRecord)
         }
       } else {
-        this.logger.debug('Connection does not exists.')
-        connectionRecord = await this.createConnection(outOfBandMessage, { handshakeProtocol, autoAcceptConnection })
+        this.logger.debug('Reuse is disabled or connection does not exist.')
+        connectionRecord = await this.createConnection(outOfBandRecord, {
+          label,
+          autoAcceptConnection,
+        })
       }
 
       if (messages) {
         this.logger.debug('Out of band message contains request messages.')
-        if (!connectionRecord.isReady) {
+        if (connectionRecord.isReady) {
+          await this.emitWithConnection(connectionRecord, messages)
+        } else {
```

Comment:
Not for PR, but now that we have an out of band record, we could update this to an event based flow. So instead of waiting for the connection to be finished, we can add an internal event listener that whenever a connection is completed checks whether there is any oob flow to be finished.